File: CobblemonMarketApi/Models/PokemonListing.cs
namespace CobblemonMarketApi.Models;

public class PokemonListing : BaseListing
{
    public string Uuid { get; set; } = "";
    public string Form { get; set; } = "";
    public int PokedexNumber { get; set; }
    public string PokemonName { get; set; } = "";
    public int Level { get; set; }
    public string Nature { get; set; } = "";
    public string Ability { get; set; } = "";
    public bool IsHiddenAbility { get; set; }
    public string Gender { get; set; } = "";
    public bool IsShiny { get; set; }
    public bool IsRadiant { get; set; }

    public string DefaultImageUrl { get; set; } = "";
    public string? CustomImageUrl { get; set; }

    public int HpIv { get; set; }
    public int AttackIv { get; set; }
    public int DefenseIv { get; set; }
    public int SpecialAttackIv { get; set; }
    public int SpecialDefenseIv { get; set; }
    public int SpeedIv { get; set; }
    public double TotalIvPercentage { get; set; }

    public ICollection<ShowcasePokemonListing> ShowcaseLinks { get; set; } = new List<ShowcasePokemonListing>();
}

File: cobblemon-market-front/cobblemon-market-front/src/app/models/pokemon-listing.ts
export interface PokemonListing {
  id: number;
  uuid: string;
  showcaseId: number;
  title: string;
  price: number;
  description: string;
  pokedexNumber: number;
  pokemonName: string;
  form?: string | null;
  level: number;
  nature: string;
  ability: string;
  isHiddenAbility: boolean;
  gender: string;
  isShiny: boolean;
  defaultImageUrl: string;
  customImageUrl?: string | null;
  hpIv: number;
  attackIv: number;
  defenseIv: number;
  specialAttackIv: number;
  specialDefenseIv: number;
  speedIv: number;
  totalIvPercentage: number;
}

/** DTO pour la création d’un listing Pokémon (aligné sur l’API) */
export interface CreatePokemonListing {
  title: string;
  price: number;
  description: string;
  pokedexNumber: number;
  pokemonName: string;
  form?: string | null;
  level: number;
  nature: string;
  ability: string;
  isHiddenAbility?: boolean;
  gender: string;
  isShiny: boolean;
  customImageUrl?: string | null;
  hpIv: number;
  attackIv: number;
  defenseIv: number;
  specialAttackIv: number;
  specialDefenseIv: number;
  speedIv: number;
}

/** DTO pour la mise a jour d'un listing Pokemon (meme structure que la creation) */
export type UpdatePokemonListing = CreatePokemonListing;

